lib/src/main/java/com/diffplug/spotless/FileSignature.java

```diff
@@ -38,70 +38,66 @@
 	 * incremental builds.
 	 */
 	@SuppressFBWarnings("SE_TRANSIENT_FIELD_NOT_RESTORED")
-	private final transient List<File> sortedFiles;
+	private final transient Collection<File> files;
 
-	@SuppressWarnings("unused")
 	private final String[] filenames;
-	@SuppressWarnings("unused")
 	private final long[] filesizes;
-	@SuppressWarnings("unused")
 	private final long[] lastModified;
 
-	public static FileSignature from(File... files) throws IOException {
-		return from(Arrays.asList(files));
+	/** Creates file signature whereas order of the files remains unchanged. */
+	public static FileSignature fromList(File... files) throws IOException {
+		return fromList(Arrays.asList(files));
 	}
 
-	public static FileSignature from(Iterable<File> files) throws IOException {
-		List<File> sortedFiles = toSortedSet(Objects.requireNonNull(files));
+	/** Creates file signature whereas order of the files remains unchanged. */
+	public static FileSignature fromList(Collection<File> files) throws IOException {
```

Comment:
The new version of the FileSignature behaves in a way like a collection. Depending on your needs you want to store a sorted set or an unsorted list. Your input [may already fulfill theses needs](https://github.com/fvgh/spotless/blob/feature/common_property_support/lib/src/main/java/com/diffplug/spotless/JarState.java#L63-L73), but [others do not](https://github.com/fvgh/spotless/blob/feature/common_property_support/lib-extra/src/main/java/com/diffplug/spotless/extra/GitAttributesLineEndings.java#L186-L190). The FileSignature takes over the job to format your input as required. So allowing the creator to specify, whether the input already fulfills (some) of the desired criteria, avoids processing time.
On the [output](https://github.com/fvgh/spotless/blob/feature/common_property_support/lib/src/main/java/com/diffplug/spotless/FileSignature.java#L84) I also don't want to loose effort, I spend already in processing the input.
Whether this optimization is worth the cluttering of the class interface, depends on the use case. 
You have a better experience  about the input (for my use case, the configuration files, it barely matters). Give me a thumbs up for keeping the approach, or a thumbs down for changing as requested.